lib/robot.js

```diff
@@ -144,19 +144,36 @@ class Robot {
       logger: log.child({installation: id})
     })
 
-    if (id) {
-      const res = await this.cache.wrap(`app:${id}:token`, () => {
-        log.trace(`creating token for installation`)
-        github.authenticate({type: 'integration', token: this.app()})
+    if (process.env.AUTH_METHOD === 'githubapp') {
```

Comment:
so tests are failing because the `github.env.AUTH_METHOD` is not being set, so the `else` defaults to OAUTH authentication.

Few options/thoughts (@bkeepers ):
- I'd prefer to default/else to GitHub_App integration, but I didn't want to do too much churn to the code. If everyone is :+1: with this PR up until this point, I can swap out `=== 'githubapp'` for `==='oauth'`. This will fix this test failure.
- However, we probably want to bootstrap these relevant tests with the proper `auth_method` being set.
- Which also means I should write a new test for authentication via oauth.
- Which should probably also write a test that involves a webhook for an oauth integration so we can test the path where the webhook doesn't have the app_id included.